### PR TITLE
Add support for phpstan 1.0

### PR DESCRIPTION
**Pull request by alexander-schranz**
We are using this package in @sulu CMS and as phpstan 1.0 was released I thought it would be great to support the new phpstan version. This PR will set the required phpstan version to 1.0. Would I think be also a good time to tag also this package with 1.0 thats why I did not add it as additional allowed version.

### CODE REVIEW

File: phpstan-without-extension-baseline.neon [hunk 1]
```diff
@@ -31,65 +31,85 @@ parameters:
 			path: test/StaticAnalysis/Test/ObjectProphecy/ProphesizeTest.php
 
 		-
-			message: "#^Call to an undefined method Prophecy\\\\Prophecy\\\\ObjectProphecy\\:\\:doubleTheNumber\\(\\)\\.$#"
-			count: 1
+			message: "#^Call to an undefined method Prophecy\\\\Prophecy\\\\ObjectProphecy\\<JanGregor\\\\Prophecy\\\\Test\\\\StaticAnalysis\\\\Src\\\\BaseModel\\>\\:\\:doubleTheNumber\\(\\)\\.$#"
+			count: 2
```
Comment (alexander-schranz) [inline phpstan-without-extension-baseline.neon:35]: I'm little bit curious why this count did increase. Not sure if I did something wrong in the baseline update, hope you can have a look at it if I did all correctly.

File: src/Type/ObjectProphecy/WillExtendOrImplementDynamicReturnTypeExtension.php [hunk 2]
```diff
@@ -62,11 +62,11 @@ public function getTypeFromMethodCall(
             return $returnType;
         }
 
-        if (0 === \count($methodCall->args)) {
+        if (0 === \count($methodCall->getArgs())) {
```
Comment (alexander-schranz) [inline src/Type/ObjectProphecy/WillExtendOrImplementDynamicReturnTypeExtension.php:65]: See Upgrade: https://github.com/phpstan/phpstan/releases/tag/1.0.0

File: src/Type/ObjectProphecy/WillExtendOrImplementDynamicReturnTypeExtension.php [hunk 3]
```diff
@@ -62,11 +62,13 @@ public function getTypeFromMethodCall(
             return $returnType;
         }
 
-        if (0 === \count($methodCall->args)) {
+        $args = $methodCall->getArgs();
```
Comment (alexander-schranz) [inline src/Type/ObjectProphecy/WillExtendOrImplementDynamicReturnTypeExtension.php:65]: See Upgrade: https://github.com/phpstan/phpstan/releases/tag/1.0.0

File: phpstan-without-extension-baseline.neon (hunk 1 above)
Comment (stof) [reply phpstan-without-extension-baseline.neon:35]: that's not only a count increase. It shows an modified message too.

Comment (alexander-schranz) [reply phpstan-without-extension-baseline.neon:35]: The object was changed by me to have 0 errors when using. The extension. Not sure but I think when the run shows 0 errors when extension is enabled and show this errors when extension not enabled we should be fine.

Comment (stof) [reply phpstan-without-extension-baseline.neon:35]: to me, that's fine indeed, based on the change you made.

**Comment by alexander-schranz** [comment]
Think @Jan0707 is not longer active on github. @localheinz would be great if you could have a look at this if you maybe got some time for it. 

If somebody want to test or use it alray add the following to your `composer.json`:

```json
   "repositories": [
      {
         "type": "vcs":
         "url": "git@github.com:alexander-schranz/phpstan-prophecy.git"
      }
   ],
   ..
   "require-dev": {
       "jangregor/phpstan-prophecy": "dev-patch-1 as 1.0.0"
   }
```

**Pull request merged**